### DISCUSSION TITLE
Is there a way we can select a specific event with `getAbiItem`

### DISCUSSION THREAD

**Discussion by zkSoju** [wevm/viem#260] [General]
I'm trying to do the following, but the ABI contains two events with different parameters both named `Transfer`.
```ts
 const event = getAbiItem({
    abi: erc20ABI,
    name: "Transfer",
  });
```

**Comment by jxom** [comment]
You can use the `args` option if you know the arguments ahead of time to discriminate the overload. If you don’t supply that, it will return the last matched ABI item.
Could make sense for this function to return an array instead, and/or accept a `params` option or something to match against param types.

**Reply by zkSoju** [reply]
I don’t know the args in advance, so it either of those solutions would be helpful. I was scratching head for a bit, since it’s the first time I’ve ran into this problem. I guess a workaround for now would be to hardcode it in

**Reply by PaulRBerg** [reply]
Just bumped into this as well. Can we open an issue for returning an array when multiple events share the same name?

**Comment by SebastienGllmt** [comment]
I'm also surprised such a footgun exists where if you `getContractEvents` on a contract with overloaded events you might just silently gets the wrong result. I assume it works this way because the two solutions are:

1. Use the function signature (which is blocked by [this missing feature](https://github.com/wevm/viem/discussions/2706))
2. Use the selector (which can't easily be statically determined by typescript since you can't calculate the hash (`toSignatureHash`) of the signature at the type level)

Neither of which is doable in Viem today. I think (1) is a lot more realistic, so I've opened an issue in abitype to add this feature: https://github.com/wevm/abitype/pull/255 with the corresponding viem PR here: https://github.com/wevm/viem/pull/2928
